support/globalconfig/infrastructure.go

```diff
@@ -82,7 +82,11 @@ func ReconcileInfrastructure(infra *configv1.Infrastructure, hcp *hyperv1.Hosted
 		if infra.Status.PlatformStatus.Azure == nil {
 			infra.Status.PlatformStatus.Azure = &configv1.AzurePlatformStatus{}
 		}
-		infra.Status.PlatformStatus.Azure.CloudName = configv1.AzurePublicCloud
+		cloudName := configv1.AzureCloudEnvironment(hcp.Spec.Platform.Azure.Cloud)
```

Comment:
These are not the same constants. I believe you'll need to add a mapping function.